### PR TITLE
*: tiny refactor config.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**
See #210 

Reduce some flags and refactor config parser.
/cc @siddontang  @huachaohuang  @overvenus

### CODE REVIEW

File: server/config.go [hunk 1]
```diff
@@ -64,21 +67,43 @@ type Config struct {
...
+	fs.StringVar(&cfg.DataDir, "data-dir", defaultDataDir, "path to the data directory (default 'default.{name}')")
+	fs.StringVar(&cfg.Host, "host", defaultHost, "host for outer traffic")
+	fs.Uint64Var(&cfg.Port, "port", defaultPort, "server port (deprecate later)")
+	fs.Uint64Var(&cfg.AdvertisePort, "advertise-port", 0, "advertise server port (deprecate later) (default '{port}')")
+	fs.Uint64Var(&cfg.HTTPPort, "http-port", defaultHTTPPort, "http port (deprecate later)")
+	fs.Uint64Var(&cfg.ClientPort, "client-port", defaultClientPort, "port for client traffic")
+	fs.Uint64Var(&cfg.AdvertiseClientPort, "advertise-client-port", 0, "advertise port for client traffic (default '{client-port}')")
+	fs.Uint64Var(&cfg.PeerPort, "peer-port", defaultPeerPort, "port for peer traffic")
+	fs.Uint64Var(&cfg.AdvertisePeerPort, "advertise-peer-port", 0, "advertise port for peer traffic (default '{peer-port}')")
+	fs.StringVar(&cfg.InitialCluster, "initial-cluser", defaultInitialCluster, "initial cluster configuration for bootstrapping (default '{name}=http://{host}:{advertise-peer-port}')")
```
Comment (siddontang) [inline server/config.go]: I think we can remove defaultInitialCluster  and leave empty here.
So if user doesn't set it, default used, but here default may be not defaultInitialCluster.

File: server/cluster.go [hunk 2]
```diff
@@ -116,18 +116,12 @@ func (c *RaftCluster) isRunning() bool {
 
 // GetConfig gets config information.
 func (s *Server) GetConfig() *Config {
-	s.cfgLock.RLock()
-	defer s.cfgLock.RUnlock()
-
 	return s.cfg.clone()
```
Comment (siddontang) [inline server/cluster.go]: why clone here?

**Comment by siddontang** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
@huachaohuang PTAL

**Comment by siddontang** [comment]
PTAL @shenli @iamxy

**Comment by huachaohuang** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**